### DISCUSSION TITLE
STARTUP HOOK for injecting into a method

### DISCUSSION THREAD

**Discussion by romanov** [dotnet/runtime#82671] [Q&A]
Hello!

In dotnet i can inject my library into any dotnet dll via `$env:DOTNET_STARTUP_HOOKS=../Hook.dll`

For now i can spawn some threads, add some additional logic outside the main app.

Is it possible to add or replace some logic to my method in original app (i know the method's assembly name, location and signature).

For example, i've a simple console app with the following method:
Is it possible to do that without any "memory injection libraries"?

```

public void Say(string message)
{
        // hook here


       // method logic
      ...
}

```

Thank you for any help.

**Comment by Xyncgas** [comment]
Yes, abandon all codes or some of your code in the original app then define them in the DLL you are injecting
Or use unsafe pointers in your original app and attempt to change them from the DLL you are injecting
You can ask FBI they do this all the time

**Reply by TonyValenti** [reply]
Or use Harmony.  Amazing library that lets you override methods in other classes.
